pkg/operators/ebpf/attach.go

```diff
@@ -145,6 +149,91 @@ func (i *ebpfInstance) attachProgram(gadgetCtx operators.GadgetContext, p *ebpf.
 		return link.AttachLSM(link.LSMOptions{
 			Program: prog,
 		})
+	case ebpf.PerfEvent:
+		perfType := uint32(unix.PERF_TYPE_SOFTWARE)
+		perfConfig := uint64(unix.PERF_COUNT_SW_CPU_CLOCK)
+		perfSampleType := uint64(unix.PERF_SAMPLE_RAW)
+		frequency := uint64(0)
+		name, ok := strings.CutPrefix(p.SectionName, perfEventPrefix)
+		// allow overriding parameters; TODO: discuss and add more? there's A LOT of options to cover
+		if ok {
+			if tmp := i.config.GetString("programs." + name + ".perf.type"); tmp != "" {
+				switch tmp {
+				case "software":
+					perfType = unix.PERF_TYPE_SOFTWARE
+				default:
+					// Try to get raw value // TODO: should we support it in this way as well?
+					val, err := strconv.ParseUint(tmp, 10, 32)
+					if err != nil {
+						return nil, fmt.Errorf("unknown perf type %q for program %q", tmp, name)
+					}
+					perfType = uint32(val)
+				}
+			}
+			if tmp := i.config.GetString("programs." + name + ".perf.config"); tmp != "" {
+				switch tmp {
+				case "count_sw_cpu_clock":
+					perfConfig = unix.PERF_COUNT_SW_CPU_CLOCK
+				default:
+					// Try to get raw value // TODO: should we support it in this way as well?
+					val, err := strconv.ParseUint(tmp, 10, 64)
+					if err != nil {
+						return nil, fmt.Errorf("unknown perf config %q for program %q", tmp, name)
+					}
+					perfConfig = val
+				}
+			}
+			if tmp := i.config.GetString("programs." + name + ".perf.sampleType"); tmp != "" {
+				switch tmp {
+				case "sample_raw":
+					perfSampleType = unix.PERF_SAMPLE_RAW
+				default:
+					// Try to get raw value // TODO: should we support it in this way as well?
+					val, err := strconv.ParseUint(tmp, 10, 64)
```

Comment:
I am not sure if we should support config and type in a generic way. I don't know if all config & type combinations are meant to be used with ebpf.

I would add those support only if we tested that it works. Otherwise, we can start with supporting just the basic PERF_SAMPLE_RAW + PERF_COUNT_SW_CPU_CLOCK.

Reply:
Hmm. Tbh, if it doesn't crash (or is harmful in other ways), I'd rather keep it than having to make potential users of those other combinations wait for at least another release cycle of IG. Seems we should invest some time to investigate other use cases.